cmd/executable_check.go

```diff
@@ -0,0 +1,95 @@
+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/dcos/dcos-go/exec"
+	"github.com/spf13/cobra"
+)
+
+var validExecutables = map[string]bool{
+	"curl":  true,
+	"wget":  true,
+	"tar":   true,
+	"git":   true,
+	"xz":    true,
+	"unzip": true,
+}
+
+// executableCmd represents the executable command
+var executableCmd = &cobra.Command{
+	Use:   "executable",
+	Short: "Check for executable/executables required to install DC/OS",
+	Long: `Check for existence of the following executable: 
+curl
+wget
+tar
+git
+xz
+unzip
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		RunCheck(context.WithValue(context.Background(), "args", args), NewExecutableCheck("DC/OS verify existence of executables"))
```

Comment:
as discussed, we will set the binary name in NewExecutableCheck constructor.
we should also setup a context with a reasonable configurable timeout and pass it to RunCheck
```
ctx, cancel := context.WithTimeout(...)
```